app-template/src/AppInitializer.tsx

```diff
@@ -33,6 +36,7 @@ class AppInitializer extends React.Component<AppInitializerProps, AppInitializer
     ).catch((_err) => {
       console.log('no token from cookie could be fetched ', _err)
     })
+    this.updateSynapseCallbackCookie()
   }
 
   componentDidUpdate(prevProps: any) {
```

Comment:
Shouldn't updateSynapseCallbackCookie also get called in componentDidUpdate? Otherwise I think it will always redirect back to the first URL put into the browser when it mounts

Reply:
Yes